### PR TITLE
Add Python documentation

### PR DESCRIPTION
**Pull request by lukesneeringer**
This PR adds [protoc-docs-plugin](https://github.com/googleapis/protoc-docs-plugin) support for the Python proto generation.

(It is also a test of whether CircleCI is working against forks now.)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -154,8 +164,4 @@ RUN mkdir -p /root/
 ADD artman-user-config-in-docker.yaml /root/.artman/config.yaml
 
 # Install artman and run the smoke test.
-RUN pip3 install googleapis-artman==0.4.4
-
-# TODO (lukesneeringer): Move smoke tests to a different venue.
-# RUN smoketest_artman.py \
-#   && rm -rf /artman/output
+RUN pip3 install googleapis-artman==0.4.5
```
Comment (ethanbao) [inline Dockerfile]: I will drop my PR which does the same change here.

File: Dockerfile [hunk 2]
```diff
@@ -124,6 +124,16 @@ RUN gem install ronn --no-ri --no-rdoc \
   && chmod a+x /usr/local/bin/php-cs-fixer \
   && cd /
 
+# Set up tools for Python codegen; these are:
+#   pandoc: an apt package that can convert text between formats
+#     (example: Markdown to Restructured Text)
+#   protoc-docs-plugin: A protoc plugin to add docstrings to the Python
+#     protoc output.cd
+RUN apt-get update \
+  && apt-get install -y pandoc \
```
Comment (ethanbao) [inline Dockerfile]: Is "pandoc" used anywhere in artman code? github codesearch doesn't tell me anything.

Comment (lukesneeringer) [inline Dockerfile]: It's used by protoc-docs-plugin. (It converts Markdown to ReST.)

File: Dockerfile (hunk 1 above)
Comment (lukesneeringer) [inline Dockerfile]: Ha! Sorry. :-p

File: Dockerfile (hunk 2 above)
Comment (ethanbao) [inline Dockerfile]: Does "--pydocstring_out={root}" flag below imply the use of protoc-docs-plugin?

Comment (lukesneeringer) [inline Dockerfile]: Exactly. The way protoc works is that `--foo_out={place}` translates to running `protoc-gen-foo` in `$PATH`.

**Pull request merged**